### PR TITLE
Add support for resources

### PR DESCRIPTION
**Pull request by pavolloffay**
Design doc: https://docs.google.com/document/d/1avSSf__R226l2b3hbcpXlYH7w6iKtXZsd9VTcpxDqng/edit#heading=h.vft382hqn4hq

Recording: https://vimeo.com/773320732/17c06cf515

Follows a sketch how the resources spec might look like in the future
```
resources:
 total:
   limits:
     cpu: 500m
     memory: 2Gi
 distribution/split:
   memory:
    ingester: 15%
    distributor: 50%
   cpu:
     ingester: 10%
     distributor: 20%
```

### CODE REVIEW

File: apis/tempo/v1alpha1/microservices_types.go [hunk 1]
```diff
@@ -94,6 +91,17 @@ type MicroservicesList struct {
 	Items           []Microservices `json:"items"`
 }
 
+// Resources defines resources configuration.
+type Resources struct {
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/microservices_types.go:95]: I am introducing this wrapper to enable us in the future add configuration of the split for the resources.

 e.g.

```yaml
cpu:
  ingester: 0.5m
  distributor: 0.2m
```

File: README.md [hunk 2]
```diff
@@ -46,6 +46,11 @@ spec:
   storage:
     secret: minio-test
   storageSize: 1Gi
+  resources:
+    resources:
```
Comment (frzifus) [inline README.md:50]:
```
  resources:
    resources:
```

Feels somehow wrong. Any ideas for other names? e.g. distribution?

File: apis/tempo/v1alpha1/microservices_types.go (hunk 1 above)
Comment (frzifus) [reply apis/tempo/v1alpha1/microservices_types.go:95]: Ive not strong opinion yet. Would it be sufficient to support only a `%` option?

Comment (pavolloffay) [reply apis/tempo/v1alpha1/microservices_types.go:95]: > Would it be sufficient to support only a % option?

@frzifus sorry I don't understand what do you mean?

**Review by frzifus — APPROVED** [review]

**Pull request merged**